tinygrad/codegen/linearizer.py

```diff
@@ -195,7 +195,7 @@ def linearize(self):
     # add a local buffer for multistage reduce. # TODO: use local alias
     if self.group_for_reduce:
       # TODO: the strides of this can be controlled
-      self.sts.append(ShapeTracker(tuple([1] * self.first_reduce + self.group_for_reduce + [1] * (self.shape_len - self.upcasted - len(self.group_for_reduce) - self.first_reduce) + [x[0] for x in self.upcasted_axis(0)])))
+      self.sts.append(ShapeTracker(tuple([1] * self.global_dims + [x for x in self.full_shape[self.global_dims:self.global_dims+self.local_dims+len(self.group_for_reduce)]] + [1] * (self.shape_len - self.upcasted - len(self.group_for_reduce) - self.first_reduce) + [x[0] for x in self.upcasted_axis(0)])))
```

Comment:
`[x for x in self.full_shape[self.global_dims:self.global_dims+self.local_dims+len(self.group_for_reduce)]]`
Should be the same as
`self.full_shape[self.global_dims:self.global_dims+self.local_dims+len(self.group_for_reduce)]` or throw a `list()` around it if `self.full_shape` is a `tuple`.

Reply:
yup thanks!  I can't hide my sloppy python from you.